### PR TITLE
SASS-1594: Adding incurred costs amount page

### PR DESCRIPTION
**Pull request by joeyymoore**
### Description

- Adding incurred costs amount page
- Linking the page to the benefits CYA page

https://jira.tools.tax.service.gov.uk/browse/SASS-1594

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

File: it/controllers/benefits/income/IncurredCostsBenefitsAmountControllerISpec.scala [hunk 1]
```diff
@@ -0,0 +1,570 @@
...
+        urlPost(pageUrl(taxYearEOY), follow = false, body = form, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+      }
+
+      "redirects to check your benefits page" in {
+        result.status shouldBe SEE_OTHER
+        result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url)
+      }
+
+      "update the paymentsOnEmployeesBehalf value to the new amount" in {
+        lazy val cyamodel = findCyaData(taxYearEOY, employmentId, userRequest).get
```
Comment (georgievh) [inline it/controllers/benefits/income/IncurredCostsBenefitsAmountControllerISpec.scala:457]: Rename cyamodel to cyaModel to fix scalastyle warning

File: it/controllers/benefits/income/IncurredCostsBenefitsAmountControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,570 @@
...
+      }
+
+      "redirect to the reimbursed costs section page" in {
+        result.status shouldBe SEE_OTHER
+//        TODO: This will update to the reimbursed costs section page when created
+        result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url)
+      }
+
+      "update the paymentsOnEmployeesBehalf value to the new amount" in {
+        lazy val cyamodel = findCyaData(taxYearEOY, employmentId, userRequest).get
```
Comment (georgievh) [inline it/controllers/benefits/income/IncurredCostsBenefitsAmountControllerISpec.scala:481]: Same as above

File: it/controllers/benefits/income/IncurredCostsBenefitsAmountControllerISpec.scala (hunk 1 above)
Comment (joeyymoore) [reply it/controllers/benefits/income/IncurredCostsBenefitsAmountControllerISpec.scala]: updated

File: it/controllers/benefits/income/IncurredCostsBenefitsAmountControllerISpec.scala (hunk 2 above)
Comment (joeyymoore) [reply it/controllers/benefits/income/IncurredCostsBenefitsAmountControllerISpec.scala]: updated

**Pull request merged**